Obtain scaling of three subimages UPPER, MIDDLE, LOWER based on given…

vSebas opened the pull request:
… ratio

vSebas committed:
Obtain scaling of three subimages UPPER, MIDDLE, LOWER based on given ratio

ivanacollg commented on src/test/data_augmentation_test.cpp:
```diff
@@ -21,6 +21,8 @@ int main( int argc, char** argv )
    //Apply average filter
    data.AverageFilter(kernel_size);
+   //Apply scaling
+   data.Scaling_ROI(1); 
```
declara los parameros de la funcion como variables al inicio del codigo

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -35,6 +35,44 @@ void DataAugmentation::AverageFilter(const int &kernel){
+    cv::Size size;
+    cv::Point offset;
+    // Width and  height of ROI
+    int w_ = std::round(ratio*in_.cols);
```
renombra variables w_ y h_ sin guin bajo porque son variables locales de la funcion. Nombrarlas con nombre mas descriptivo. Ejemplo: width y height

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -35,6 +35,44 @@ void DataAugmentation::AverageFilter(const int &kernel){
+                                                    0, 0, CV_INTER_LINEAR);
+      cv::namedWindow("Cropped image",CV_WINDOW_NORMAL); 
+      cv::imshow("Cropped image",croppedImage_);
+      if(i == 0){
```
Cambia estos tres ifs por un int switch

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -35,6 +35,44 @@ void DataAugmentation::AverageFilter(const int &kernel){
+      cv::namedWindow("Cropped image",CV_WINDOW_NORMAL); 
+      cv::imshow("Cropped image",croppedImage_);
+      if(i == 0){
+      cv::imwrite("../../imgs/Scale_UPPER_x_"+std::to_string(ratio)+".png",
```
no le pongas mayusculas a los paths de salida

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -35,6 +35,44 @@ void DataAugmentation::AverageFilter(const int &kernel){
+      size.width = w_;
+      //Crop the ROI from original image
+      cv::Rect ROI(offset, size);
+      cv::Mat croppedImage_ = image_(ROI);  
```
cropped image es una variable local, por lo que no lleva guin bajo. Usa out_ el lugar de cropped image

vSebas replied:
Done

ivanacollg commented on src/data_augmentation.cpp:
```diff
@@ -35,6 +35,44 @@ void DataAugmentation::AverageFilter(const int &kernel){
+
+void DataAugmentation::Scaling_ROI(const float ratio){
+  if(ratio < 1){
+    cv::Mat image_ = in_.clone();
```
cambia el nombre de image_ a temp ya que es una matriz temporaria en lo que llegas a out_. Y no lleva gion bajo porque es variable de la funcion, no de la clase

vSebas replied:
Done

ivanacollg reviewed (changes requested):
Arregla los merge conflics y modifica los headers de los files como coautor y fecha modificada

vSebas committed:
Made these changes:
Variable passed as parameter in testing unit
Renamed some local variable and objects
Added switch case statement
Changed the name of the output paths
Changed croppedImage_ for out_

ivanacollg commented:
Arregla merge conflicts

vSebas committed:
Updated brief and added parameter comment

vSebas committed:
Added Ivana's mail

vSebas committed:
Merge branch 'master' into feature/scaling

ivanacollg committed:
Merge branch 'master' of github.com:vanttec/data_augmentation into feature/scaling
